### DISCUSSION TITLE
Trying to Run the Audio test program to generate Sawtooth on ESP32-A1S Devkit)

### DISCUSSION THREAD

**Discussion by j-raghavan** [grame-cncm/faust#1004] [General]
Hello!

I am following https://faustdoc.grame.fr/tutorials/esp32/ To create a test application on my ESP32 (ESP32-A1S Devkit) But running into weird issue. 

Getting compilation issue with PIN_CTRL

```faust-test/main/FaustSawtooth.cpp:1474:23: error: 'PIN_CTRL' was not declared in this scope
 1474 |             REG_WRITE(PIN_CTRL, 0xFFFFFFF0);
v5.2.1/esp-idf/components/soc/esp32/include/soc/soc.h:42:13: note: in expansion of macro 'ASSERT_IF_DPORT_REG'
   42 |             ASSERT_IF_DPORT_REG((_r), REG_WRITE);  
```

I am using Faust Version 
```
[main]$faust --version
FAUST Version 2.70.3
Embedded backends: 
   DSP to C
   DSP to C++
   DSP to Cmajor
   DSP to Codebox
   DSP to CSharp
   DSP to DLang
   DSP to FIR
   DSP to Interpreter
   DSP to Java
   DSP to JAX
   DSP to Julia
   DSP to JSFX
   DSP to LLVM IR
   DSP to old C++
   DSP to Rust
   DSP to WebAssembly (wast/wasm)
Build with LLVM version 17.0.6
Copyright (C) 2002-2024, GRAME - Centre National de Creation Musicale. All rights reserved. 
```

I am new to ESP32 and am very interested in getting Faust to work on my Devboard. I may be doing something very silly. But can't seem to figure out what the issue is. Could someone help out here. 

Here are the Generated files. 

[FaustSawtooth.cpp.txt](https://github.com/grame-cncm/faust/files/14734700/FaustSawtooth.cpp.txt)
[FaustSawtooth.h.txt](https://github.com/grame-cncm/faust/files/14734702/FaustSawtooth.h.txt)
[main.cpp.txt](https://github.com/grame-cncm/faust/files/14734703/main.cpp.txt)
[WM8978.cpp.txt](https://github.com/grame-cncm/faust/files/14734705/WM8978.cpp.txt)
[WM8978.h.txt](https://github.com/grame-cncm/faust/files/14734706/WM8978.h.txt)

Thank you!
Jay

**Comment by sletz** [comment]
Better come on[ Faust Discord](https://faust.grame.fr/community/help/#faust-on-discord) and use the #esp32 channel there.
